tests/openid_connect/pro_connect/tests.py

```diff
@@ -703,6 +706,49 @@ def test_callback_ft_users_unknown_safir_already_in_org(self, client, pro_connec
             ],
         )
 
+    # FIXME (alaurent) Remove in a month (old states expiry)
+    @respx.mock
+    def test_old_state_compat(self, client, pro_connect):
+        user = PrescriberFactory(
+            **dataclasses.asdict(ProConnectPrescriberData.from_user_info(pro_connect.oidc_userinfo))
+        )
+        authorize_params = {
+            "user_kind": "employer",
+            "next_url": "/next_url",
+            "user_email": user.email,
+            "register": False,
+        }
+
+        # Calling this view is mandatory to start a new session.
+        authorize_url = f"{reverse('pro_connect:authorize')}?{urlencode(authorize_params)}"
+        response = client.get(authorize_url)
+        assert response.url.startswith(constants.PRO_CONNECT_ENDPOINT_AUTHORIZE)
+
+        # Edit the state
+        state = ProConnectState.objects.get()
+        del state.data["enforce_kind"]  # Old state didn't have it
+        state.save()
+
+        token_json = {"access_token": "access_token", "token_type": "Bearer", "expires_in": 60, "id_token": "123456"}
+        respx.post(constants.PRO_CONNECT_ENDPOINT_TOKEN).mock(return_value=httpx.Response(200, json=token_json))
+
+        user_info = OIDC_USERINFO.copy()
+        # Put a issued at in the future to ensure we don't check it
+        user_info = user_info | {
```

Comment:
J'ai peut-être raté un truc mais _a priori_ pas besoin du `.copy()` vu qu'il n'y a pas de mutation de l'objet :
```suggestion
        # Put a issued at in the future to ensure we don't check it
        user_info = OIDC_USERINFO | {
```